src/main/java/org/jboss/pnc/deliverablesanalyzer/CacheProvider.java

```diff
@@ -65,9 +96,44 @@ private static void ensureConfigurationDirectoryExists() throws IOException {
         }
     }
 
+    /**
+     * Return the appropriate cache manager depending on the infinispan mode
+     *
+     * @return cache manager
+     * @throws IOException something went wrong
+     */
     @Produces
-    @SuppressWarnings("deprecation")
-    public DefaultCacheManager initCaches() throws IOException {
+    public BasicCacheContainer initCaches() throws IOException {
+        switch (infinispanMode) {
+            case EMBEDDED:
+                LOGGER.info("Using Embedded Infinispan cache");
+                return setupEmbeddedCacheManager();
+            case REMOTE:
+                LOGGER.info("Using Remote Infinispan cache");
+                return setupDistributedCacheManager();
+            default:
+                throw new RuntimeException("This infinispan mode has no cache manager");
+        }
+    }
+
+    public void close(@Disposes BasicCacheContainer cacheManager) {
+        if (cacheManager instanceof Cloneable) {
```

Comment:
Cloneable? Not Closeable?

Reply:
OH RIGHT! My bad! and thanks for catching this!!! I'm embarassed by this mistake :(